### ISSUE TITLE
Only 0.0 - 1.0 objectives?

### ISSUE THREAD

**Issue by mccartney** [OpenSLO/OpenSLO#24]
>     target: numeric [0.0, 1.0) # budget target for given objective of the SLO
>    timeSliceTarget: numeric (0.0, 1.0] # required only when budgetingMethod is set to TimeSlices

This assumes the values will be only between 0% and 100% which seems to match the typical availability service level indicators. And generally a good practice. Although I saw other kind of SLIs in real-life.

Relevant SRE book chapter:
https://sre.google/workbook/implementing-slos/
which generally advocates for defining the 0%-100% ones. And even introduce the notions of "SLI specification" and "SLI implementations" with only the latter ones matching 0%-100%.

TLDR: I suggest to open up for other kind of SLIs, not only the [0,1] ones.

**Comment by its02003** [comment]
Can you give an example of an objective target where the availability would be something outside of 0-100%?

**Comment by mccartney** [comment]
> Can you give an example of an objective target where the availability would be something outside of 0-100%?

Not for availability. I have only ever saw availability being 0-100%.

But I thought this definition of `timeSliceTarget` is not only for availability SLIs, but other SLIs too. If so, examples of SLIs which are not 0-100% are:

- p95 of HTTP request processing time per second
- number of failed customer transactions per minute

**Comment by Maixy** [comment]
>p95 of HTTP request processing time per second
This should still be a 0-100% target.  With TimeSliceTarget you'd be counting up the number of intervals per rolling window where the p95 was < your target divided by the total number of intervals per window. 
ex. `Minutes where request processing time p95 < 500ms in the last 28d / minutes in the last 28d`
This results in a percentage between 0 and 100

> number of failed customer transactions per minute
This should presumably include a threshold that's acceptable, ex. 10 failed transactions per minute is 🆗  but beyond that isn't acceptable. 
With that threshold, each minute gets evaluated separately, and we get the similar equation to the above example:
`Minutes in the last 28d where the number of failed transactions were less than 10 / total minutes in the last 28d`

**Comment by nieomylnieja** [comment]
I'm closing this as it seems @Maixy has answered the question and there seems to be no more objections.
